hspec-core/src/Test/Hspec/Core/Golden.hs

```diff
@@ -0,0 +1,75 @@
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE ExistentialQuantification #-}
+
+module Test.Hspec.Core.Golden
+  ( Golden(..)
+  , GoldenResult(..)
+  , defaultGolden
+  , runGolden
+  )
+  where
+
+import           System.Directory ( createDirectoryIfMissing
+                                  , doesFileExist
+                                  )
+
+
+-- | Golden tests parameters
+
+data Golden str =
+  forall. Eq str =>
+    Golden {
```

Comment:
Specifically, I think you can use the `Eq` constraint to `runGolden` to get rid of `ExistentialQuantification`.